packages/medusa/src/services/product.ts

```diff
@@ -501,6 +505,26 @@ class ProductService extends TransactionBaseService {
         })
       )
 
+      if (variants) {
+        const toCreate = variants.map((variant) => {
```

Comment:
todo: I am getting an error creating variants without providing variant options to `ProductService.create`. I know this is somewhat of an edge case, but we should handle it gracefully. 

My suggestion would be to default to empty variant options in case they are not passed.